dds/DCPS/DiscoveryBase.h

```diff
@@ -2752,8 +2752,8 @@ namespace OpenDDS {
       }
 
       virtual void update_subscription_locators(DDS::DomainId_t domainId,
-                                                const RepoId& partId,
-                                                const RepoId& subId,
+                                                const RepoId partId,
+                                                const RepoId subId,
```

Comment:
Many GitHub Actions warnings here - looks like this can't change without updating the related (base or derived) class

Reply:
This was another data race, but thinking about it afterwards it seems odd to me that this fixed it. I was just trying to prevent the lower level methods from reaching back and doing memcmp directly against the datareaderimpl member... but the read would still happen to make a copy of the RepoId. I'll take a look at the warnings.